index.js

```diff
@@ -41,13 +41,17 @@ module.exports = function(content, filename, context) {
     var options = {filename: filename, lineOffset: 0, displayErrors: true};
     var compiledWrapper = vm.runInThisContext(wrapper, options);
 
+    var moduleKeysCount = Object.keys(sandbox.module).length;
+    var exportKeysCount = Object.keys(sandbox.module.exports).length;
     compiledWrapper.apply(sandbox.exports, args);
 
     var result;
-    if(sandbox.module.exports['__42__'] && Object.keys(sandbox.module.exports).length === 1) {
+    if(
+        Object.keys(sandbox.module.exports).length === exportKeysCount &&
+        Object.keys(sandbox.module).length === moduleKeysCount
+    ) {
         result = context ? vm.runInNewContext(content, context) : vm.runInThisContext(content);
     } else {
-        delete sandbox.module.exports['__42__'];
```

Comment:
I'm looking for tests. I don't see where we checking case with pushing data to `exports` object.

Can you recheck it?
```js
nodeEval('exports.lolka = 123').shouldEqual({lolka: 123});
```